### PR TITLE
Create spotxBidAdapter2.js

### PR DESCRIPTION
**Pull request by jmonbillard**

### CODE REVIEW

File: spotxBidAdapter2.js [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+          mimes: mimes
+        }
+      };
+
+      if (utils.getBidIdParameter('price_floor', bid.params) != '') {
+        spotxImp.bidfloor = utils.getBidIdParameter('price_floor', bid.params);
+      }
+
+      if (utils.getBidIdParameter('start_delay', bid.params.video) != '') {
+        spotxImp.startdelay = 0 + Boolean(utils.getBidIdParameter('start_delay', bid.params.video));
```
Comment (strider820) [inline spotxBidAdapter2.js]: I believe this belongs under video

File: spotxBidAdapter2.js [hunk 2]
```diff
@@ -0,0 +1,362 @@
+import * as utils from 'src/utils';
+import { Renderer } from 'src/Renderer';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER, VIDEO } from 'src/mediaTypes';
+
+const BIDDER_CODE = 'spotx2';
+const URL = '//search.spotxchange.com/openrtb/2.3/dados/';
+const VERSION = '2.3';
```
Comment (danielmcgraw) [inline spotxBidAdapter2.js]: ORTB_VERSION so we don't confuse it with the adaptor or dados version

File: spotxBidAdapter2.js [hunk 3]
```diff
@@ -0,0 +1,362 @@
+import * as utils from 'src/utils';
+import { Renderer } from 'src/Renderer';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER, VIDEO } from 'src/mediaTypes';
+
+const BIDDER_CODE = 'spotx2';
```
Comment (danielmcgraw) [inline spotxBidAdapter2.js]: why is this spotx2?

File: spotxBidAdapter2.js [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+const URL = '//search.spotxchange.com/openrtb/2.3/dados/';
+const VERSION = '2.3';
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['spotx2'],
```
Comment (danielmcgraw) [inline spotxBidAdapter2.js]: is 'spotx2' the name of this in prebid?

File: spotxBidAdapter2.js (hunk 1 above)
Comment (danielmcgraw) [inline spotxBidAdapter2.js]: spotxImp.video.startdelay

File: spotxBidAdapter2.js [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+    }
+
+    let userExt = {};
+
+    // Add GDPR flag and consent string
+    if (bidderRequest && bidderRequest.gdprConsent) {
+      userExt.consent = bidderRequest.gdprConsent.consentString;
+
+      if (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean') {
+        payload.regs = {
```
Comment (danielmcgraw) [inline spotxBidAdapter2.js]: I believe you mean requestPayload here

File: spotxBidAdapter2.js [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, bidderRequest) {
+    const bidResponses = [];
+    serverResponse = serverResponse.body;
```
Comment (danielmcgraw) [inline spotxBidAdapter2.js]: Do we really want to blow away serverResponse here or should we create a new var?

File: spotxBidAdapter2.js [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+    script.setAttribute('data-spotx_ad_unit', 'instream');
+    script.setAttribute('data-spotx_ad_skippable', '0');
+    script.setAttribute('data-spotx_custom_skin', '1');
+  } else {
+    script.setAttribute('data-spotx_ad_unit', 'incontent');
+    script.setAttribute('data-spotx_collapse', '0');
+  }
+
+  script.setAttribute('data-spotx_autoplay', '1');
+  // script.setAttribute('data-spotx_blocked_autoplay_override_mode', '1');
```
Comment (danielmcgraw) [inline spotxBidAdapter2.js:348]: There a reason this floating comment is left in?

File: spotxBidAdapter2.js [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+  script.setAttribute('data-spotx_video_slot_can_autoplay', '1');
+
+  if (bid.renderer.config.inIframe && window.document.getElementById(bid.renderer.config.inIframe).nodeName == 'IFRAME') {
+    var rawframe = window.document.getElementById(bid.renderer.config.inIframe);
```
Comment (danielmcgraw) [inline spotxBidAdapter2.js]: Why are we using var here all of the sudden?

File: spotxBidAdapter2.js [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+
+function outstreamRender(bid) {
+  if (bid.renderer.config.outstreamFunction != null && typeof bid.renderer.config.outstreamFunction === 'function') {
+    bid.renderer.config.outstreamFunction();
+    return;
+  }
+
+  window.console.log('[SPOTX][renderer] Handle SpotX outstream/inbanner renderer');
+
+  var script = window.document.createElement('script');
```
Comment (danielmcgraw) [inline spotxBidAdapter2.js]: var?

File: spotxBidAdapter2.js [hunk 10]
```diff
@@ -0,0 +1,362 @@
+import * as utils from 'src/utils';
```
Comment (danielmcgraw) [inline spotxBidAdapter2.js:1]: Review the use of let, const, and var in the file. They all seem to be used and not exactly in the right context all the time.

**Review by danielmcgraw — CHANGES_REQUESTED** [review]

File: spotxBidAdapter2.js (hunk 7 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js:348]: This can be removed. It's a comment I added for my testing phase. The outstreamRender JS function is used to handle the outstream rendering inside prebid adaptor and not outside. It's an non-mandatory function this since can be handled outside using a callback function in the json config of prebid.

Of course we should removed any non-used block-code when we will push it to prebid github.

File: spotxBidAdapter2.js (hunk 8 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js]: Just changed it to let.
Could you please give me some guidance for the use of const vs. let vs. var? Thanks

File: spotxBidAdapter2.js (hunk 9 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js]: Just changed it to let.

File: spotxBidAdapter2.js (hunk 6 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js]: You are right. Just change it!

File: spotxBidAdapter2.js (hunk 5 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js]: Good catch! Just changed it.

File: spotxBidAdapter2.js (hunk 1 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js]: Correct. Just made the change.

File: spotxBidAdapter2.js (hunk 4 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js]: Just changed it to ‘spotx'

File: spotxBidAdapter2.js (hunk 2 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js]: ok

File: spotxBidAdapter2.js (hunk 3 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js]: changed to 'spotx'

File: spotxBidAdapter2.js (hunk 10 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js:1]: done

File: spotxBidAdapter2.js (hunk 8 above)
Comment (danielmcgraw) [reply spotxBidAdapter2.js]: https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75

**Comment by danielmcgraw** [comment]
For more on var/let/const check out 
https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75

TLDR
Never use var
Use const if the variable never gets reassigned 
Use let if the variable gets reassigned at some point

File: spotxBidAdapter2.js [hunk 11]
```diff
@@ -0,0 +1,363 @@
...
+  if (bid.renderer.config.outstreamFunction != null && typeof bid.renderer.config.outstreamFunction === 'function') {
+    bid.renderer.config.outstreamFunction();
+    return;
+  }
+
+  window.console.log('[SPOTX][renderer] Handle SpotX outstream/inbanner renderer');
+
+  let script = window.document.createElement('script');
+  script.type = 'text/javascript';
+  script.src = '//search.spotxchange.com/js/' + bid.channel_id + '.js';
```
Comment (strider820) [inline spotxBidAdapter2.js:318]: let's use js.spotx.tv/easi/v1/ here instead

File: spotxBidAdapter2.js [hunk 12]
```diff
@@ -0,0 +1,363 @@
...
+                  return utils.logMessage('SpotX outstream video loaded event');
+                },
+                ended: function ended() {
+                  utils.logMessage('SpotX outstream renderer video event');
+                }
+              });
+            } catch (err) {
+              utils.logWarn('Prebid Error calling setRender or setEve,tHandlers on renderer', err);
+            }
+            bid.renderer = renderer;
```
Comment (strider820) [inline spotxBidAdapter2.js:269]: This will only be used if the renderer isn't assigned by the pub... I think that's ok, but we should test that

File: spotxBidAdapter2.js (hunk 11 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js:318]: done

File: spotxBidAdapter2.js (hunk 12 above)
Comment (jmonbillard) [reply spotxBidAdapter2.js:269]: Sure. How do you want to proceed with testing?
